php-checks/src/main/java/org/sonar/php/checks/ParameterSequenceCheck.java

```diff
@@ -0,0 +1,103 @@
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.symbols.Symbol;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.FunctionTree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.expression.NameIdentifierTree;
+import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import javax.annotation.CheckForNull;
+
+@Rule(key = "S2234")
+public class ParameterSequenceCheck extends PHPVisitorCheck {
+
+  public static final String MESSAGE = "Parameters to \"%s\" have the same names but not the same order as the method arguments.";
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    // only check method and function calls expect constructors with more than 1 argument
+    if (!tree.getParent().is(Tree.Kind.NEW_EXPRESSION) && tree.arguments().size() > 1) {
+      Symbol symbol = getDeclarationSymbol(tree);
+      if (symbol != null && symbol.declaration() != null) {
+        checkParameterSequence(tree, (NameIdentifierTree) symbol.declaration());
+      }
+    }
+
+    super.visitFunctionCall(tree);
+  }
+
+  @CheckForNull
+  private Symbol getDeclarationSymbol(FunctionCallTree tree) {
+    ExpressionTree callee = tree.callee();
+
+    if (callee.is(Tree.Kind.NAMESPACE_NAME)) {
+      return context().symbolTable().getSymbol(((NamespaceNameTree) callee).name());
+    } else if(isVerifiableObjectMemberAccess(callee)) {
+      return context().symbolTable().getSymbol(((MemberAccessTree) callee).member());
+    } else if(isVerifiableClassMemberAccess(callee)) {
+      return context().symbolTable().getSymbol(((MemberAccessTree) callee).member());
+    }
+
+    return null;
+  }
+
+  private static boolean isVerifiableObjectMemberAccess(ExpressionTree tree) {
+    return tree.is(Tree.Kind.OBJECT_MEMBER_ACCESS)
+      && ((MemberAccessTree) tree).member().is(Tree.Kind.NAME_IDENTIFIER)
+      && ((MemberAccessTree) tree).object().is(Tree.Kind.VARIABLE_IDENTIFIER)
+      && ((VariableIdentifierTree) ((MemberAccessTree) tree).object()).text().equals("$this");
+  }
+
+  private static boolean isVerifiableClassMemberAccess(ExpressionTree tree) {
+    return tree.is(Tree.Kind.CLASS_MEMBER_ACCESS)
+      && ((MemberAccessTree) tree).member().is(Tree.Kind.NAME_IDENTIFIER)
+      && ((((MemberAccessTree) tree).object().is(Tree.Kind.NAMESPACE_NAME)
+          && ((NamespaceNameTree) ((MemberAccessTree) tree).object()).fullName().equals("self"))
+        || (((MemberAccessTree) tree).object().is(Tree.Kind.NAME_IDENTIFIER)
+          && ((NameIdentifierTree) ((MemberAccessTree) tree).object()).text().equals("static")));
```

Comment:
Creating intermediate variables would make the code more readable to me, for example for `((MemberAccessTree) tree).object()`.